monkey/monkey_island/cc/next_ui/src/app/(protected)/layout.tsx

```diff
@@ -1,6 +1,10 @@
+'use client';
 import '../../styles/globals.scss';
 import MainLayout from '@/layouts/main-layout/MainLayout';
+import useAuthRedirection from '@/app/(protected)/_lib/useAuthRedirection';
 
 export default function AppLayout({ children }: { children: React.ReactNode }) {
+    useAuthRedirection();
```

Comment:
I'm a bit surprised that this works because my understanding was that Next.js only renders the layouts once, but perhaps my understanding is wrong